modules/justpremiumBidAdapter.js

```diff
@@ -1,274 +1,186 @@
-import bidfactory from 'src/bidfactory';
-import bidmanager from 'src/bidmanager';
-import adloader from 'src/adloader';
-import * as utils from 'src/utils';
-import adaptermanager from 'src/adaptermanager';
-
-const CONSTANTS = require('src/constants.json');
-
-const JustpremiumAdapter = function JustpremiumAdapter() {
-  const top = window.top;
-  let d;
-  let bids;
-  let cookieLoaded = false;
-  let adManagerLoaded = false;
-  let jPAM;
-  let dConfig;
-  let toLoad;
-  let server;
-
-  function isCrossOriginIframe() {
-    try {
-      return !top.document;
-    } catch (e) {
-      return true;
+import { registerBidder } from 'src/adapters/bidderFactory'
+
+const BIDDER_CODE = 'justpremium'
+const ENDPOINT_URL = top.document.location.protocol + '//pre.ads.justpremium.com/v/2.0/t/xhr'
```

Comment:
`top` should generally be in a try/catch block, or use `utils.getTopWindowLocation()` to safely get `location` here